ext/zlib/zlib.c

```diff
@@ -816,10 +816,10 @@ static bool zlib_create_dictionary_string(HashTable *options, char **dict, size_
 						*++ptr = zval_get_string(cur);
 						if (!*ptr || ZSTR_LEN(*ptr) == 0 || EG(exception)) {
 							if (*ptr) {
-								efree(*ptr);
+								zend_string_release(*ptr);
 							}
 							while (--ptr >= strings) {
-								efree(ptr);
+								zend_string_release(*ptr);
 							}
```

Comment:
I believe this can be adjusted to a do-while loop for consistency with the NUL check case below. Or perhaps this can be unified into a single place with a `goto`.

Reply:
Indeed, since `zval_get_string()` always returns a `zend_string`, we can simplify.

Regarding goto see https://xkcd.com/292/ ;)

Reply:
I prefer a well-placed goto over duplicating error handling all over the place. In fact it's already a little messy with the `EG(exception)` check reusing the first `if()` for the error handling only.

Reply:
In my opinion, it's much better than it was, and I'm fine with. Maybe we get a third opinion. :)